Floofbot/Modules/Logging.cs

```diff
@@ -177,226 +135,297 @@ public async Task Toggle()
         // events handling
         public class EventHandlingService{
 
-            SqliteConnection dbConnection;
-            public EventHandlingService()
+            FloofDataContext _floofDb;
+            public EventHandlingService(FloofDataContext floofDb)
             {
-                dbConnection = new SqliteConnection(new SqliteConnectionStringBuilder
-                {
-                    DataSource = "botdata.db"
-                }.ToString());
-                dbConnection.Open();
+                _floofDb = floofDb;
             }
-            public async Task<ITextChannel> GetChannel(string tableName, Discord.IGuild guild)
+            public async Task<ITextChannel> GetChannel(string eventName, Discord.IGuild guild)
             {
-                // gets a channel based on the tablename (the type of logger)
-                string sqlGetChannel = $"SELECT {tableName} FROM Logger WHERE ServerID = {guild.Id} LIMIT 1"; // check state
-                using (SqliteCommand command = new SqliteCommand(sqlGetChannel, dbConnection))
+                // TODO: Find a better algorithm for this. Hardcoding event names is :(

```

Comment:
Instead of having to add a `case` for each event type, you can instead try to get the channel ID by treating the `serverConfig` like a map. See [here](https://stackoverflow.com/questions/11395315) on getting property values from arbitrary key names.

Reply:
I agree on the validEvents part and will update the code to reflect this. I am confused by the first part, could you give an example of what you mean? I don't think I have ever used "maps"

Reply:
Maps being hashmaps or something like Python dictionaries, or JS objects if you're familiar. Essentially, the goal is to have some arbitrary string key and have it map to a value on the `serverConfig` object.

Say we have an arbitrary `eventName` as an input:
```cs
string eventName = "MessageDeletedChannel";
var serverConfig = _floofDb.LogConfigs.Find(guild.Id);
System.Reflection.PropertyInfo propertyInfo = serverConfig.GetType().GetProperty(eventName);
ulong logChannel = (ulong)(propertyInfo.GetValue(serverConfig, null));
```

(Turns out the `null` in `GetValue` is [not for setting a default](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.propertyinfo.getvalue?view=netcore-3.1#System_Reflection_PropertyInfo_GetValue_System_Object_System_Object___). However, for the same reasoning as the second bit in the original comment, I think it should be fine.)

Reply:
Oh I see what you are doing. Yeah, I think I was trying to just compensate for the possibility of a malformed input, but since its a function used by the developers only, I can see why this solution works.